addons/stock_account/models/product.py

```diff
@@ -135,8 +203,37 @@ class ProductProduct(models.Model):
         help="Technical field to correctly show the currently selected company's currency that corresponds "
              "to the totaled value of the product's valuation layers")
     stock_valuation_layer_ids = fields.One2many('stock.valuation.layer', 'product_id')
-    valuation = fields.Selection(related="categ_id.property_valuation", readonly=True)
-    cost_method = fields.Selection(related="categ_id.property_cost_method", readonly=True)
+    cost_method = fields.Selection(
+        string="Cost Method",
+        selection=const.COST_METHOD,
+        compute='_compute_cost_method',
+    )
+    valuation = fields.Selection(
+        string="Valuation",
+        selection=const.VALUATION,
+        compute='_compute_valuation',
+    )
+
+    @api.depends("categ_id.property_cost_method")
+    def _compute_cost_method(self):
+        for product in self:
+            product.cost_method = (
+                product.categ_id.property_cost_method
+                or (product.company_id or self.env.company).cost_method
+            )
+
+    @api.depends('categ_id.property_valuation')
+    def _compute_valuation(self):
+        product_with_catogory = self.filtered('categ_id')
+        has_group_stock_accounting_automatic = self.env.user.has_group(
+            'stock_account.group_stock_accounting_automatic'
+        )
+        if has_group_stock_accounting_automatic:
+            (self - product_with_catogory).valuation = 'real_time'
+        else:
+            (self - product_with_catogory).valuation = 'manual_periodic'
+        for product in product_with_catogory:
+            product.valuation = product.categ_id.property_valuation
```

Comment:
Same comments as for the computed fields on the templates ;)
Btw, is it really useful to redefine those fields on `product.product` model?
They could be either removed (and fall back on the template through the inherits), or be computed based on the template value instead.

Reply:
Done!
I thought variant category can be different from template :slightly_frowning_face: